File: reference/CodeExamples/dotNet/CSharpExamples/Code/disconnectedobj/idexamples/IIdExample.cs
using Db4objects.Db4o;
using Db4objects.Db4o.Config;

namespace Db4oDoc.Code.DisconnectedObj.IdExamples
{

    public interface IIdExample<TId>
    {
        TId IdForObject(object obj, IObjectContainer database);
        object ObjectForID(TId idForObject, IObjectContainer database);
        void Configure(IEmbeddedConfiguration configuration);
        void RegisterEventOnContainer(IObjectContainer container);
    }
}

File: reference/CodeExamples/dotNet/VisualBasicExamples/Code/disconnectedobj/idexamples/IIdExample.vb
Imports Db4objects.Db4o
Imports Db4objects.Db4o.Config

Namespace Db4oDoc.Code.DisconnectedObj.IdExamples

    Public Interface IIdExample(Of TId)
        Function IdForObject(ByVal obj As Object, ByVal database As IObjectContainer) As TId
        Function ObjectForID(ByVal idForObject As TId, ByVal database As IObjectContainer) As Object
        Sub Configure(ByVal configuration As IEmbeddedConfiguration)
        Sub RegisterEventOnContainer(ByVal container As IObjectContainer)
    End Interface
End Namespace
